apps/frontend/src/app/3_organisms/TransactionStepDialog/components/TransactionStep/TransactionStep.tsx

```diff
@@ -49,52 +56,54 @@ export const TransactionStep: FC<TransactionStepProps> = ({
   status,
   transaction,
   config,
+  receipt,
   gasPrice,
   updateConfig,
   isLoading,
 }) => {
+  const { request, title, subtitle } = transaction;
   const [token, setToken] = useState<TokenDetailsData | undefined>();
 
-  useEffect(() => {
-    findContract(transaction.contract.address).then(result => {
-      if (result.group === 'tokens') {
-        getTokenDetailsByAddress(transaction.contract.address).then(setToken);
-      }
-    });
-  }, [transaction.contract.address]);
+  const isTransaction = useMemo(() => isTransactionRequest(request), [request]);
```

Comment:
Sorry but my advice was a bit incorrect - `isTransaction` returns `boolean` but since `isTransactionRequest` is a type guard, you cannot memoize it. We have 2 options - rewrite it to be a simple function that returns `boolean` or use it directly without `useMemo` (only in callbacks, you can use this variable in the render). I'd lean towards the second option since I don't like the fact you now need to cast everything with `as SignTransactionRequest`.